### PR TITLE
[BACKLOG-1156] As a Sqoop Export user I need to select named clusters

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

**Comment by e-cuellar** [comment]
@mdamour1976 Please Sir!

File: src/org/pentaho/di/job/entries/sqoop/SqoopConfig.java [hunk 1]
```diff
@@ -175,29 +180,47 @@
...
-          this, getClass().getMethod( "getNamenodePort" ), getClass().getMethod( "setNamenodePort", String.class ) ) );
-      items.add( new ArgumentWrapper( JOBTRACKER_HOST, BaseMessages.getString( getClass(), "JobtrackerHost.Label" ),
-          false, this, getClass().getMethod( "getJobtrackerHost" ), getClass().getMethod( "setJobtrackerHost",
-              String.class ) ) );
-      items.add( new ArgumentWrapper( JOBTRACKER_PORT, BaseMessages.getString( getClass(), "JobtrackerPort.Label" ),
-          false, this, getClass().getMethod( "getJobtrackerPort" ), getClass().getMethod( "setJobtrackerPort",
-              String.class ) ) );
-      items.add( new ArgumentWrapper( BLOCKING_EXECUTION, BaseMessages
-          .getString( getClass(), "BlockingExecution.Label" ), false, this, getClass().getMethod(
+        items.add( new ArgumentWrapper( NAMENODE_HOST, BaseMessages.getString( getClass(), "NamenodeHost.Label" ), false,
```
Comment (mdamour1976) [inline src/org/pentaho/di/job/entries/sqoop/SqoopConfig.java]: @e-cuellar Can you fix the formatting on these lines?

File: src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java [hunk 2]
```diff
@@ -561,6 +675,9 @@ protected void toggleQuickMode( boolean quickMode ) {
   protected void setUiMode( JobEntryMode mode ) {
     switch ( mode ) {
       case QUICK_SETUP:
+        if ( !this.selectedNamedCluster.equals( USE_ADVANCED_OPTIONS_CLUSTER ) && !suppressEventHandling) {
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java]: @e-cuellar formatting

File: src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java [hunk 3]
```diff
@@ -176,10 +198,85 @@ protected void createBindings( S config, XulDomContainer container, BindingFacto
...
+
+  public void setNamedClusters( AbstractModelList <NamedCluster> namedClusters ) {
+    this.namedClusters = namedClusters;
+  }
+
+  public void setSelectedNamedCluster( NamedCluster namedCluster ) {
+    this.selectedNamedCluster = namedCluster;
+    if ( !suppressEventHandling ) {
+      if ( namedCluster != null && !namedCluster.equals( CHOOSE_AVAILABLE_CLUSTER )
+          && !namedCluster.equals( USE_ADVANCED_OPTIONS_CLUSTER )) {
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java]: @e-cuellar formatting

File: src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java [hunk 4]
```diff
@@ -176,10 +198,85 @@ protected void createBindings( S config, XulDomContainer container, BindingFacto
...
+    @SuppressWarnings("unchecked") XulMenuList<NamedCluster>
+        namedClusterMenu = (XulMenuList<NamedCluster>) container.getDocumentRoot().getElementById( "named-clusters" ); //$NON-NLS-1$
+    try {
+      String cn = config.getClusterName();
+      if ( cn != null ) {
+        NamedCluster namedCluster = NamedClusterUIHelper.getNamedCluster( cn );
+        namedClusterMenu.setSelectedItem( namedCluster );
+        setSelectedNamedCluster( namedCluster );
+      } else if ( cn == null && (config.getNamenodeHost() != null || config.getNamenodePort() != null ||
+          config.getJobtrackerHost() != null || config.getJobtrackerPort() != null )) {
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java]: @e-cuellar formatting

File: src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java [hunk 5]
```diff
@@ -74,16 +81,23 @@
...
   private DatabaseItem selectedDatabaseConnection;
   private DatabaseDialog databaseDialog;
+  protected NamedCluster selectedNamedCluster;
 
   // Flag to indicate we shouldn't handle any events. Useful for preventing unwanted synchronization during
   // initialization
   // or other user-driven events.
   protected boolean suppressEventHandling = false;
 
+  protected HadoopClusterDelegate ncDelegate = new HadoopClusterDelegate( Spoon.getInstance() );;
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/job/entries/sqoop/AbstractSqoopJobEntryController.java]: @e-cuellar Extra semicolon

File: src/org/pentaho/di/job/entries/sqoop/SqoopConfig.java [hunk 6]
```diff
@@ -635,4 +618,68 @@ public void setMode( String mode ) {
...
+    return clusterName;
+  }
+
+  public void setClusterName( String clusterName ) {
+    this.clusterName = clusterName;
+  }
+
+  public String getNamenodeHost() {
+      NamedCluster namedCluster = getNamedCluster();
+      if( namedCluster != null ) {
```
Comment (mdamour1976) [inline src/org/pentaho/di/job/entries/sqoop/SqoopConfig.java]: @e-cuellar this file has a few if( formatting issues.

**Pull request merged**